enclave/src/lib.rs

```diff
@@ -265,22 +256,11 @@ pub unsafe extern "C" fn get_state(
 		None => vec!(0),
 	};
 
-	// split the extrinsic_slice at the length of the encoded extrinsic
-	// and fill the right side with whitespace
-	let (left, right) = value_slice.split_at_mut(value_vec.len());
-	left.clone_from_slice(&value_vec);
-	//FIXME: now implicitly assuming we pass unsigned integer vecs, not strings terminated by 0x20
-	//FIXME: we should really pass an Option<Vec<u8>>
-	right.iter_mut().for_each(|x| *x = 0x00);
-	//right.iter_mut().for_each(|x| *x = 0x20);
-	sgx_status_t::SGX_SUCCESS
-}
+//	//FIXME: now implicitly assuming we pass unsigned integer vecs, not strings terminated by 0x20
+//	//FIXME: we should really pass an Option<Vec<u8>>
```

Comment:
I do not really get, what is meant here

Reply:
what I meant is: It is now hard-coded that the return value from `get_state()` is an unsigned integer. A generic solution would allow generic data types to be returned (with variable byte size). This function should be agnostic to type and just use a SCALE-encoded `Option<Vec<u8>>`